Python/ceval.c

```diff
@@ -265,12 +265,14 @@ void
 Py_SetRecursionLimit(int new_limit)
 {
     PyInterpreterState *interp = _PyInterpreterState_GET();
+    _PyEval_StopTheWorld(interp);
     interp->ceval.recursion_limit = new_limit;
     for (PyThreadState *p = interp->threads.head; p != NULL; p = p->next) {
```

Comment:
This still needs a `HEAD_LOCK()`. Looping over the list requires the lock, even with a stop the world pause. The code that adds and removes thread states from the linked list doesn't require the GIL/active thread state.

Reply:
Ah I missed that, the version in main branch uses `_Py_FOR_EACH_TSTATE_BEGIN` which acquires that lock before iterating.